mobly/controllers/android_device_lib/adb.py

```diff
@@ -121,27 +139,27 @@ def _exec_cmd(self, args, shell, timeout):
             AdbError: The adb command exit code is not 0.
             AdbTimeoutError: The adb command timed out.
         """
+        cmd_str = ' '.join(args)
```

Comment:
This changes the way command line arguments are displayed from `['cp', 'my file', 'other file']` to `"cp my file other file"`. The change is not strictly necessary for this pull request, but if you'd like to proceed with it, I think the proper way would be:

```python
def _quote_command(command):
  return ' '.join(pipes.quote(arg) for arg in command)
```
...returning `cp 'my file' 'other file'`

Reply:
In debugging a failure caused by AdbError, I find it pretty common to try to run the same cmd manually to see the results. Having this as a list makes pasting the cmd difficult.

So the motivation here is so we can directly copy the cmd from the error msg.

`pipes.quote` is used for constructing cmd to be passed to pipes, whereas we purely want something for displaying. So I don't think `pipes` is necessary. Also `pipes` only works on POSIX, which is not good as we need to support windows as well.

Is there an alternative?

Reply:
I have explained why correct quoting is necessary, with it you actually can copy/paste the command unmodified, without fearing for any unusual characters in the command. ` ` `;` `$` `*` `(` `)` `{` `}` `'` `"` - any of these will make the command invalid. And not just for the purposes of copy pasting, but even seeing what's going on.

I believe `pipes.quote` is available on Windows.
https://github.com/python/cpython/blob/2.7/Lib/pipes.py#L267
https://github.com/python/cpython/blob/3.6/Lib/pipes.py#L65

Reply:
Done